ee/desktop/runner/runner.go

```diff
@@ -212,40 +231,91 @@ func (r *DesktopUsersProcessesRunner) Interrupt(interruptError error) {
 	}
 }
 
+// Update handles control server updates for the desktop subsystem
+// This includes data related to the runner itself, and the user processes
 func (r *DesktopUsersProcessesRunner) Update(data io.Reader) error {
-	var desktopStatus client.DesktopUserStatus
+	var desktopStatus DesktopUserStatus
 	if err := json.NewDecoder(data).Decode(&desktopStatus); err != nil {
 		return fmt.Errorf("failed to decode desktop user control data: %w", err)
 	}
 
-	if desktopStatus.Status == "" {
-		return fmt.Errorf("empty desktop status")
+	// Regardless, we will write the menu data out to a file that can be grabbed by
+	// any desktop user processes, either when they refresh, or when they are spawned.
+	r.writeMenuFile(desktopStatus.Menu)
```

Comment:
this returns an err, should we log it or something?

Reply:
Good catch - returning this error now